frontend/packages/process-editor/test/EventListeners.test.ts

```diff
@@ -0,0 +1,179 @@
+import { EventListeners } from './EventListeners';
+
+describe('EventListeners', () => {
+  describe('add', () => {
+    it('Adds a listener to the given event', () => {
+      const eventListeners = new EventListeners<{ event: () => void }>();
+      const fun = jest.fn();
+      const eventName = 'event';
+
+      eventListeners.add(eventName, fun);
+      eventListeners.triggerEvent(eventName);
+
+      expect(fun).toHaveBeenCalledTimes(1);
+    });
+
+    it('Supports adding multiple listeners to the same event', () => {
+      const eventListeners = new EventListeners<{ event: () => void }>();
+      const fun1 = jest.fn();
+      const fun2 = jest.fn();
+      const eventName = 'event';
+
+      eventListeners.add(eventName, fun1);
+      eventListeners.add(eventName, fun2);
+      eventListeners.triggerEvent(eventName);
+
+      expect(fun1).toHaveBeenCalledTimes(1);
+      expect(fun2).toHaveBeenCalledTimes(1);
+    });
+
+    it('Supports adding listeners to multiple events', () => {
+      const eventListeners = new EventListeners<Record<'event1' | 'event2', () => void>>();
+      const event1Fun = jest.fn();
+      const event2Fun = jest.fn();
+      const event1Name = 'event1';
+      const event2Name = 'event2';
+
+      eventListeners.add(event1Name, event1Fun);
+      eventListeners.add(event2Name, event2Fun);
+      eventListeners.triggerEvent(event1Name);
+      eventListeners.triggerEvent(event2Name);
+
+      expect(event1Fun).toHaveBeenCalledTimes(1);
+      expect(event2Fun).toHaveBeenCalledTimes(1);
+    });
+  });
```

Comment:
Could this be a reasonable case to test?

```
it('Supports adding same function to multiple events', () => {
      const eventListeners = new EventListeners<Record<'event1' | 'event2', () => void>>();
      const event1Fun = jest.fn();
      const event1Name = 'event1';
      const event2Name = 'event2';

      eventListeners.add(event1Name, event1Fun);
      eventListeners.add(event2Name, event1Fun);
      eventListeners.triggerEvent(event1Name);
      eventListeners.triggerEvent(event2Name);

      expect(event1Fun).toHaveBeenCalledTimes(2);
    });
```

Reply:
This is a plausible use case indeed. I'll add it, thanks!